### PR TITLE
Oc 17103 service proxy issues

### PR DESCRIPTION
**Pull request by sidux**
Check https://github.com/OpenClassrooms/ServiceProxy/pull/6/files for description.

### CODE REVIEW

File: OpenClassroomsServiceProxyBundle.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ public function build(ContainerBuilder $container)
     {
         parent::build($container);
 
-        $container->addCompilerPass(new ServiceProxyPass(), PassConfig::TYPE_AFTER_REMOVING);
+        $container->addCompilerPass(new ServiceProxyPass(), PassConfig::TYPE_AFTER_REMOVING, 1);
```
Comment (sidux) [inline OpenClassroomsServiceProxyBundle.php:20]: Increase the compiler pass priority to make execute before UseCaseBundle

File: Services/Proxy/ConfigurationFactory.php [hunk 2]
```diff
@@ -29,6 +30,8 @@ public function create($cacheDir, $dumpAutoload = false)
                 new FileWriterGeneratorStrategy(new FileLocator($cacheDir))
             );
             spl_autoload_register($configuration->getProxyAutoloader());
+        } else {
+            $configuration->setGeneratorStrategy(new TmpFileGeneratorStrategy($cacheDir));
```
Comment (sidux) [inline Services/Proxy/ConfigurationFactory.php:34]: Using a file generation strategy for dev env, check https://github.com/OpenClassrooms/ServiceProxy/pull/6/files

Comment (romainkuzniak) [reply Services/Proxy/ConfigurationFactory.php:34]: did you think about what we talk?

Comment (sidux) [reply Services/Proxy/ConfigurationFactory.php:34]: yes, i didn't get any noticeable change, proxies generation takes 0.0007s in both strategies `eval` and `tmp`, it's probably due to ssd high r/w speed

**Comment by sidux** [comment]
we'll treat this again, post php8.1 and service proxy rewrite

**Pull request closed without merging**